integration_tests/e2e/agree-plan.cy.ts

```diff
@@ -0,0 +1,70 @@
+describe('Agree plan', () => {
+  beforeEach(() => {
+    cy.createSentencePlan().then(planDetails => {
+      cy.wrap(planDetails).as('plan')
+      cy.openSentencePlan(planDetails.oasysAssessmentPk)
+      cy.get('.govuk-button').contains('Agree plan').click()
```

Comment:
Once the gating of Agree is in place the tests will start failing because the Plan doesn't have any Goals which contain Steps. I suggest you use the DataGenerator to set that up here, and also create unhappy path tests for 1) plan with no goals and 2) plan with goals but no steps.

Reply:
I think this should be done when gating of Agree is in place. Its based on current state of main branch.

Reply:
OK